metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/iceberg/IcebergTableHandler.java

```diff
@@ -99,7 +125,15 @@ public IcebergTableUtil(final ConnectorContext connectorContext) {
      * @return iceberg table
      */
     public Table getIcebergTable(final QualifiedName tableName, final String tableMetadataLocation) {
-        return new IcebergMetastoreTables(tableMetadataLocation).load(tableName.toString());
+        this.icebergTableCriteria.checkCriteria(tableName, tableMetadataLocation);
+        final long start = this.registry.clock().wallTime();
+        log.debug("Loading icebergTable {} from {}", tableName, tableMetadataLocation);
+        final Table table = new IcebergMetastoreTables(tableMetadataLocation).load(tableName.toString());
+        final long duration = registry.clock().wallTime() - start;
+        log.info("Time taken to getIcebergTable {} is {} ms", tableName, duration);
```

Comment:
Same here. If the above logic fails, the metrics will not be reported.

Reply:
Will move to finally block.